File: Chapter_01_Introduction/Getting_Started_The_Easy_Problems/Super_Easy/11498-Division_of_Nlogonia/Main.java
/*
 * Problem 11498 Division of Nlogonia
 * Java version 6/8/2016
 */

import java.io.PrintStream;
import java.util.Scanner;

public class Main {
    static final Scanner input = new Scanner(System.in);
    static final PrintStream output = System.out;

    public static void main(String[] args) {

        StringBuffer results = new StringBuffer();

        while (true) {
            int numberOfQueries = input.nextInt();
            if (numberOfQueries == 0) {
                break;
            }

            int n = input.nextInt();
            int m = input.nextInt();

            for (int i = 0; i < numberOfQueries; ++i) {
                int x = input.nextInt();
                int y = input.nextInt();

                if (x == n || y == m) {
                    results.append("divisa\n");
                } else if (x > n) {
                    if (y > m) {
                        results.append("NE\n");
                    } else {
                        results.append("SE\n");
                    }
                } else {
                    if (y > m) {
                        results.append("NO\n");
                    } else {
                        results.append("SO\n");
                    }
                }
            }
        }
        System.out.print(results.toString());
    }
}

File: Chapter_01_Introduction/Getting_Started_The_Easy_Problems/Super_Easy/11498-Division_of_Nlogonia/main.cpp
/*
 * Problem 11498 Division of Nlogonia
 */
#include <iostream>
#include <sstream>

int main() {
    std::ios::sync_with_stdio(0);
    std::cin.tie(0);

    std::ostringstream output;

    int testCases;
    while (std::cin >> testCases && testCases != 0) {
        int m, n;
        std::cin >> m >> n;
        while (testCases--) {
            int x, y;
            std::cin >> x >> y;
            if (x == m || y == n) {
                output << "divisa\n";
            } else if (x > m) {
                output << (y > n ? "NE" : "SE") << '\n';
            } else {
                output << (y > n ? "NO" : "SO") << '\n';
            }
        }
    }
    std::cout << output.str();
}
